luskan/src/test/java/org/opencadc/luskan/StorageLocationConverterTest.java

```diff
@@ -0,0 +1,194 @@
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import ca.nrc.cadc.tap.TapQuery;
+import ca.nrc.cadc.tap.schema.TapSchema;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class StorageLocationConverterTest {
+    private static final Logger log = Logger.getLogger(StorageLocationConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.loadTapSchema();
+
+    @Test
+    public void testSchemaName() {
+        String query = "select column_name from tap_schema.columns11";
+        String expected = "SELECT column_name FROM tap_schema.columns11";
+        doTest(query, expected, null);
+    }
+
+    @Test
+    public void testSchemaAndTableName() {
+        String query = "select lastModified from inventory.storagesite";
+        String expected = "SELECT lastModified FROM inventory.storagesite";
+        doTest(query, expected, null);
+    }
+
+    @Test
+    public void testSelectWithoutStorageLocation() {
+        String query = "select uri from inventory.artifact";
+        String expected = "SELECT uri FROM inventory.artifact";
+        doTest(query, expected, null);
+    }
+
+    @Test
+    public void testSelectWithStorageLocation() {
+        String query = "select uri from inventory.artifact";
+        String expected = "SELECT uri FROM inventory.artifact WHERE (inventory.artifact.storagelocation_storageid IS NOT NULL)";
+        doTest(query, expected, "true");
+    }
+
+    @Test
+    public void testSelectColumns() {
+        String query = "select uri, lastmodified from inventory.artifact";
+        String expected = "SELECT uri, lastmodified FROM inventory.artifact WHERE (inventory.artifact.storagelocation_storageid IS NOT NULL)";
+        doTest(query, expected, "true");
+    }
+
+    @Test
+    public void testSelectWithTableAlias() {
+        String query = "select uri from inventory.artifact a";
+        String expected = "SELECT uri from inventory.artifact AS a WHERE (a.storagelocation_storageid IS NOT NULL)";
+        doTest(query, expected, "true");
+    }
+
+    @Test
+    public void testSelectWithTableJoin() {
+        String query = "select a.uri from inventory.artifact a join inventory.artifact b on a.id = b.id";
+        String expected = "SELECT a.uri from inventory.artifact AS a JOIN inventory.artifact AS b on a.id = b.id WHERE (a.storagelocation_storageid IS NOT NULL)";
```

Comment:
The expected query is wrong here - there should also be a 
```
(b.storagelocation_storageid IS NOT NULL)
```
in there. See comment in StorageLocationCnverter

Reply:
Updated the unit test with the extra constraint.